Formula/nlohmann-json.rb

```diff
@@ -0,0 +1,86 @@
+class NlohmannJson < Formula
+  desc "JSON for Modern C++"
+  homepage "https://github.com/nlohmann/json"
+  url "https://github.com/nlohmann/json/archive/v3.6.1.tar.gz"
+  sha256 "80c45b090e40bf3d7a7f2a6e9f36206d3ff710acfa8d8cc1f8c763bb3075e22e"
+  head "https://github.com/nlohmann/json.git", :branch => "develop"
+
+  depends_on "cmake" => :build
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-DJSON_BuildTests=OFF", *std_cmake_args
+      system "make", "install"
+    end
+  end
+
+  test do
+    (testpath/"test.cpp").write <<~EOS
+      #include <nlohmann/json.hpp>
+
+      using nlohmann::json;
+
+      int main() {
+        // create an empty structure (null)
+        json j;
+
+        // add a number that is stored as double (note the implicit conversion of j to an object)
+        j["pi"] = 3.141;
+
+        // add a Boolean that is stored as bool
+        j["happy"] = true;
+
+        // add a string that is stored as std::string
+        j["name"] = "Niels";
+
+        // add another null object by passing nullptr
+        j["nothing"] = nullptr;
+
+        // add an object inside the object
+        j["answer"]["everything"] = 42;
+
+        // add an array that is stored as std::vector (using an initializer list)
+        j["list"] = { 1, 0, 2 };
+
+        // add another object (using an initializer list of pairs)
+        j["object"] = { {"currency", "USD"}, {"value", 42.99} };
+
+        // instead, you could also write (which looks very similar to the JSON above)
```

Comment:
Let's simplify the test. Remove comments, let's keep only a few objects in the json (say `pi`, `name`, `list` and `object`). Serialise the resulting json and match the output of that program (using `shell_output` and `assert_match`), so we exercise I/O as well.